### DISCUSSION TITLE
The AMQP operation was interrupted

### DISCUSSION THREAD

**Discussion by ilhas15** [rabbitmq/rabbitmq-server#11094] [Questions] [answered]
Hello Everyone I just get trouble when publish message using .NET , i got error message like this
`Already closed: The AMQP operation was interrupted: AMQP close-reason, initiated by Library, code=541, text='Unexpected Exception', classId=0, methodId=0, cause=System.IO.IOException: Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host..
 ---> System.Net.Sockets.SocketException (10054): An existing connection was forcibly closed by the remote host.
   at System.Net.Sockets.NetworkStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   --- End of inner exception stack trace ---
   at System.Net.Sockets.NetworkStream.Read(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.BufferedStream.ReadByteSlow()
   at RabbitMQ.Client.Impl.InboundFrame.ReadFrom(Stream reader, Byte[] frameHeaderBuffer, ArrayPool`1 pool, UInt32 maxMessageSize)
   at RabbitMQ.Client.Impl.SocketFrameHandler.ReadFrame()
   at RabbitMQ.Client.Framing.Impl.Connection.MainLoopIteration()
   at RabbitMQ.Client.Framing.Impl.Connection.MainLoop()`
Has anyone ever faced this kind of trouble? 
What caused it and how did you resolve it?

**Comment by michaelklishin** [comment, marked as answer]
> Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host

is fairly specific. The server has closed this TCP connection. We cannot know why but [node logs should be able to tell that](https://www.rabbitmq.com/docs/logging#logged-events).

.NET client supports [automatic connection recovery](https://www.rabbitmq.com/client-libraries/dotnet-api-guide#recovery) which a no-brainer for applications that consume. Publishers will have to [use publisher confirms](https://www.rabbitmq.com/docs/publishers#data-safety) on top of that.

**Reply by michaelklishin** [reply]
"Forcibly closed" mean mean that the node was shutting down, which means all TCP connections to it will be closed by the kernel. Nodes log when they are asked to shut down using `rabbitmqctl shutdown` and such.
